tests-integration/tests/region_migration.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use client::{DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME};
+use common_config::wal::KafkaConfig;
+use common_config::WalConfig;
+use common_meta::key::{RegionDistribution, TableMetadataManagerRef};
+use common_meta::peer::Peer;
+use common_meta::wal::kafka::KafkaConfig as MetaKafkaConfig;
+use common_meta::wal::WalConfig as MetaWalConfig;
+use common_query::Output;
+use common_telemetry::info;
+use common_test_util::temp_dir::create_temp_dir;
+use frontend::error::Result as FrontendResult;
+use frontend::instance::Instance;
+use futures::future::BoxFuture;
+use meta_srv::error::Result as MetaResult;
+use meta_srv::metasrv::SelectorContext;
+use meta_srv::procedure::region_migration::RegionMigrationProcedureTask;
+use meta_srv::selector::{Namespace, Selector, SelectorOptions};
+use servers::query_handler::sql::SqlQueryHandler;
+use session::context::{QueryContext, QueryContextRef};
+use store_api::storage::RegionId;
+use table::metadata::TableId;
+use tests_integration::cluster::{GreptimeDbCluster, GreptimeDbClusterBuilder};
+use tests_integration::test_util::{
+    check_output_stream, get_test_store_config, run_test_with_kafka_wal, StorageType,
+};
+use uuid::Uuid;
+
+const TEST_TABLE_NAME: &str = "migration_target";
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_region_migration_fs() {
+    common_telemetry::init_default_ut_logging();
+    run_test_with_kafka_wal(|endpoints| {
+        Box::pin(async move { test_region_migration(StorageType::File, endpoints).await })
+    })
+    .await
+}
+
+pub async fn test_region_migration(store_type: StorageType, endpoints: Vec<String>) {
+    let cluster_name = "test_region_migration";
+    let peer_factory = |id| Peer {
+        id,
+        addr: "127.0.0.1:3001".to_string(),
```

Comment:
I don't like the hardcode of `addr`.

Reply:
We need a placeholder to make the world work🥹. Otherwise, the channel manager will throw an invalid URL error.